File: modules/foundation/src/main/java/com/projecthub/base/submission/infrastructure/event/store/EventStore.java
package com.projecthub.base.submission.infrastructure.event.store;

import com.projecthub.base.submission.domain.event.SubmissionEvent;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class EventStore {
    private final List<SubmissionEvent> events = new ArrayList<>();

    public void save(SubmissionEvent event) {
        events.add(event);
    }

    public List<SubmissionEvent> getEvents() {
        return new ArrayList<>(events);
    }
}

File: src/main/kotlin/com/projecthub/events/EventStore.kt
package com.projecthub.events

import org.springframework.data.domain.Sort
import java.util.*

interface EventStore {
    suspend fun save(event: DomainEvent)
    suspend fun saveAll(events: List<DomainEvent>)
    suspend fun findByAggregateId(aggregateId: UUID, sort: Sort = Sort.by("occurredOn")): List<DomainEvent>
    suspend fun findByType(type: String): List<DomainEvent>
}
